src/AstTransforms.cpp

```diff
@@ -527,44 +530,54 @@ bool RemoveBooleanConstraintsTransformer::transform(AstTranslationUnit& translat
                 bool containsTrue = false;
                 bool containsFalse = false;
 
+                // Check if aggregator body contains booleans.
                 for (AstLiteral* lit : aggr->getBodyLiterals()) {
                     if (auto* bc = dynamic_cast<AstBooleanConstraint*>(lit)) {
-                        bc->isTrue() ? containsTrue = true : containsFalse = true;
+                        if (bc->isTrue()) {
+                            containsTrue = true;
+                        } else {
+                            containsFalse = true;
+                        }
                     }
                 }
 
+                // Only keep literals that aren't boolean constraints
                 if (containsFalse || containsTrue) {
-                    // Only keep literals that aren't boolean constraints
                     auto replacementAggregator = std::unique_ptr<AstAggregator>(aggr->clone());
-                    replacementAggregator->clearBodyLiterals();
+                    std::vector<std::unique_ptr<AstLiteral>> newBody;
 
                     bool isEmpty = true;
 
                     // Don't bother copying over body literals if any are false
                     if (!containsFalse) {
                         for (AstLiteral* lit : aggr->getBodyLiterals()) {
-                            // Don't add in 'true' boolean constraints
+                            // Don't add in boolean constraints
                             if (dynamic_cast<AstBooleanConstraint*>(lit) == nullptr) {
                                 isEmpty = false;
-                                replacementAggregator->addBodyLiteral(
-                                        std::unique_ptr<AstLiteral>(lit->clone()));
+                                newBody.push_back(std::unique_ptr<AstLiteral>(lit->clone()));
                             }
                         }
+
+                        // If the body is still empty and the original body contains true add it now.
+                        if (containsTrue && isEmpty) {
+                            newBody.push_back(std::make_unique<AstBinaryConstraint>(BinaryConstraintOp::EQ,
```

Comment:
Is there a reason we're not using `AstBoolean(true)`?

Reply:
From what I can see, the `AstTranslator` currently does not handle `AstBoolean` (test `semantic/bool` fails). This should be easy to change (we could translate `true` -> `1 = 1` at the later stage), but I would leave it for a follow-up.